src/migrate/index.js

```diff
@@ -105,9 +107,25 @@ export default class Migrator {
   // dependent on the migration config settings.
   _ensureTable() {
     var table = this.config.tableName;
+    var lockTable = this._getLockTableName(table);
     return this.knex.schema.hasTable(table)
       .then((exists) => {
         if (!exists) return this._createMigrationTable(table);
+      })
+      .then(() => {
+        return this.knex.schema.hasTable(lockTable)
+          .then((exists) => {
```

Comment:
This `.then` could be in the same level with main promise flow... I don't see any reason here to use nested thens. So end result would be something like: 
```js
.then(() => this.knex.schema.hasTable(lockTable))
.then((exist) => !exist && this._createMigrationLockTable(lockTable))
.then(() => {
  return this.knex(lockTable)...
}
```